### ISSUE TITLE
JSONAPI v6.0.0

### ISSUE THREAD

**Issue by alecgorge** [alecgorge/jsonapi#426]
Here are the things on my list for 6.0.0:

- [ ] Fix all outstanding 1.8 bugs
- [ ] Put netty on port 20059
- [ ] Deprecate port 20059 usage. The join port is so much more consistent.
- [ ] Eliminate the socket server on port 20060.
- [ ] Remove the standalone WebSocket server on port 20061. Access via WebSocket should be via `http://<server>:<joinport>/api/2/websocket` or `http://<server>:20059/api/2/websocket`. Remove all non-netty WebSocket components.
- [ ] Remove legacy v1 API entirely
- [ ] Move *every* API method to be in Java method + annotation
- [ ] Rewrite documentation generator to pull from Java annotations (spin up server, call API method to dump them all?)

Do you have any more suggestions?

**Comment by koesie10** [comment]
Allow nested methods?

**Comment by alecgorge** [comment]
What do you mean by nested method calls?

**Comment by koesie10** [comment]
Nested method calls as decribed in #129

**Comment by alecgorge** [comment]
I'm eliminating the JSON definition of API methods entirely so that will become moot. Hopefully that isn't an issue

**Comment by koesie10** [comment]
I don't actually need it myself, it was just something I saw coming up earlier in #129 and was interested in.

**Comment by krisdb2009** [comment]
Just want to say, This is awsome! Keep it up!

**Comment by krisdb2009** [comment]
Oh, also Not sure if its been fixed in latest builds but, sometimes the plugin will say "Not a valid group (insert group here)" on startup. But not all the time. Just sometimes... And, if you use /jsonapi reload it will reload the plugin but the method for grabbing chat in game will return duplicated chat in the array.

**Comment by alecgorge** [comment]
Yeah, users/groups are really screwed up right now. Would anyone be sad if I rewrote it to be json?

I would be able to make it oodles more reliable than the freaking yaml based configs...

**Comment by krisdb2009** [comment]
Go for it. It takes 5 seconds to set up the groups anyhow.

**Comment by krisdb2009** [comment]
Also, could there be a way in the future to compare usernames to uuid's. Like a way to  track them

**Comment by maxlehot1234** [comment]
Any update for jsonapi ??? Thank you

**Comment by ArcasCZ** [comment]
Hello. What about porting to Sponge? Is it real?
